tensorflow/python/training/saver.py

```diff
@@ -1070,6 +1070,10 @@ def save(self, sess, save_path, global_step=None, latest_filename=None,
     if not isinstance(sess, session.SessionInterface):
       raise TypeError("'sess' must be a Session; %s" % sess)
 
+    # Note a few lines above save_path was set to os.path.dirname(save_path)
```

Comment:
You have a good point here. I moved the check up and changed it to work on unmodified `os.path.dirname(save_path)`.
I think it's more explicit than setting `save_path = os.path.dirname(save_path)`, which is just bound to cause confusion sooner or later.